sdk_lib/sdk_container_common.sh

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# This file contains common functions used across SDK container scripts.
+
+#
+# globals
+#
+sdk_container_common_versionfile="sdk_container/.repo/manifests/version.txt"
+
+# Common "echo" function
+
+function yell() {
+    echo -e "\n###### $@ ######"
+}
+# --
+
+
+# Creates the Flatcar build / SDK version file. We assume that both
+#  Flatcar OS version and SDK version are the same.
+# Must be called from the script root.
+
+function create_versionfile() {
+    local version="$1"
+
+    yell "Writing versionfile '$sdk_container_common_versionfile', setting version '$version'."
+
+	cat >"$sdk_container_common_versionfile" <<EOF
+FLATCAR_VERSION=$version
+FLATCAR_VERSION_ID=$version
+FLATCAR_BUILD_ID=""
+FLATCAR_SDK_VERSION=$version
+EOF
+}
+# --
+
+
+function get_version_from_versionfile() {
+    if [ -f "$sdk_container_common_versionfile" ]; then
+        echo "$(source "$sdk_container_common_versionfile"; echo "$FLATCAR_SDK_VERSION")"
+    fi
+}
+# --
+
+
+# Check out ebuild repos tag/branch/commit-ish.
+# Must be called from the script root.
+
+function checkout_ebuild_repos() {
+    local version="$1"
+
+    yell "Checking out portage-stable, coreos-overlay tags"
+    for repo in portage-stable coreos-overlay; do
+		cd "sdk_container/src/third_party/$repo"
+		git fetch --all
```

Comment:
The git-submodule part is the one thing that I don't like about this. Can we try to avoid them please, they don't work well to track dependencies at all.
We need the repos during container build and then later during usage. At container build we can definitely clone, during usage I would continue relying on a slimmed down repo manifest.

Reply:
I understand your concern - submodules are often rather painful to use. Let's put this up for discussion for the time being.

The motivation here is to make `flatcar-scripts` the authoritative (top-level) repo for any given SDK and OS image version - removing the need for manifests altogether.
A usage example would be to clone `flatcar-scripts`, check out a specific version tag (say, 2969.0.0), and automatically get the respective `portage-stable`and `coreos-overlay` states.
Submodules help with automating this since the `flatcar-scripts` tag will have submodules point to the respective commits of `portage-stable`and `coreos-overlay`.
An alternative (though softer) version of the above would be to also tag both `portage-stable`and `coreos-overlay` repos with 2969.0.0 at release time.

Reply:
I've updated the "Simplify SDK sources" roadmap item with a design proposal to better define the goals: https://github.com/kinvolk/Flatcar/issues/440

Reply:
I think submodules are ok as long as the scripting takes care of using the right commit and not a stale checkout. We could implement the same again with text files but I don't think this helps much.
We need a single place to tie the commits of the three repos together. If we do it here as submodules, it allows us to start automatic builds from PRs.